sds_data_manager/lambda_code/SDSCode/instruments/l1b_codice.py

```diff
@@ -24,15 +24,16 @@ def lambda_handler(event: dict, context):
 
     # Get the environment variables
     bucket = os.environ["S3_BUCKET"]
-    prefix = os.environ["S3_KEY_PATH"]
+    prefixes = json.loads(os.environ["S3_KEY_PATH"])
 
     # Retrieves objects in the S3 bucket under the given prefix
-    try:
-        s3 = boto3.client("s3")
-        object_list = s3.list_objects_v2(Bucket=bucket, Prefix=prefix)["Contents"]
-        logger.info(f"Object list: {object_list}")
-    except KeyError:
-        logger.warning("No files present.")
+    for prefix in prefixes:
+        try:
+            s3 = boto3.client("s3")
```

Comment:
```suggestion
    s3 = boto3.client("s3")
    for prefix in prefixes:
        try:
```
Move the client up so we only get one client outside the loop and not request a new one each time. I'm not sure if it would also make sense to get the bucket right away outside the loop and then filter on prefixes after that?